### ISSUE TITLE
Vine: Function Hash Calculation Excluding Preamble Imports

### ISSUE THREAD

**Issue by JinZhou5042** [cooperative-computing-lab/cctools#3569]
Currently, the way to create the hash of a library is:
```python
functions_hash = package_serverize.generate_functions_hash(function_list)
```
Which is inside of function `create_library_from_functions(...)`

The unique hash is generated based on the content of functions, excluding preamble import statements. 

Therefore, if we want to update a previously installed library by altering only the preamble import statements, it actually results in an identical hash and does not lead to the creation of a new library.

Should this be a problem?

**Comment by dthain** [comment]
@BarrySlyDelgado what do you think?

**Comment by BarrySlyDelgado** [comment]
This is an issue. I have to familiarize myself with the library however, with this change, it is probably best to create a hash using the whole network code with the addition of the preamble imports now.

**Comment by dthain** [comment]
Just FYI this particular hash is not the TaskVine hash for caching objects at remote workers.

Instead, it is a hash to compute the uniqueness of the library itself, so that Poncho does not have to constantly rebuild the library (and environment) when its constituents have not changed.  That way, the library can be cached locally at the manager and not constantly rebuilt.

**Comment by JinZhou5042** [comment]
Resolved in #3837
